js/models/purchase/Item.js

```diff
@@ -33,23 +35,56 @@ export default class extends BaseModel {
     };
   }
 
+  get constraints() {
+    return {
+      minPaymentAddressLength: 1,
+      maxPaymentAddressLength: 256,
+    };
+  }
+
   validate(attrs) {
     const errObj = this.mergeInNestedErrors({});
     const addError = (fieldName, error) => {
       errObj[fieldName] = errObj[fieldName] || [];
       errObj[fieldName].push(error);
     };
 
-    if (attrs.quantity === 'undefined') {
-      addError('quantity', app.polyglot.t('orderModelErrors.mustHaveQuantity'));
-    } else if (!Number.isInteger(attrs.quantity)) {
-      addError('quantity', app.polyglot.t('orderModelErrors.quantityMustBeNumber'));
-    } else if (attrs.quantity < 1) {
-      addError('quantity', app.polyglot.t('orderModelErrors.noItems'));
+    if (!this.isCrypto) {
+      if (!Number.isInteger(attrs.quantity)) {
+        addError('quantity', app.polyglot.t('purchaseItemModelErrors.quantityMustBeInteger'));
+      } else if (attrs.quantity < 1) {
+        addError('quantity', app.polyglot.t('purchaseItemModelErrors.mustHaveQuantity'));
+      }
+
+      if (typeof attrs.paymentAddress !== 'undefined') {
+        addError('paymentAddress', 'The payment address should only be provide on ' +
+          'crypto listings');
+      }
+    } else {
+      const inventory = this.inventory();
+
+      if (typeof attrs.quantity !== 'number') {
+        addError('quantity', app.polyglot.t('purchaseItemModelErrors.quantityMustBeNumeric'));
+      } else if (attrs.quantity <= 0) {
```

Comment:
They're doing different checks.

Non crypto:
- ensure integer
- must be greater than or equal to 1

Crypto:
- ensure number
- must be greater than 0
- cannot exceed inventory

Reply:
Right. I was noticing that if the number has to be an integer, then < 1 is the same as <= 0, since there are no integers between 1 and 0. So checking <= 0 could be done once for both.

ie: check attrs.quantity <=0 before if (!this.isCrypto) {.

It's not a big deal, it'd be a small simplification of the code.